test/test_optim.py

```diff
@@ -2486,13 +2488,14 @@ def test_averaged_model_exponential_use_state_dict(self):
 
         def avg_fn(p_avg, p, n_avg):
             return alpha * p_avg + (1 - alpha) * p
-        averaged_dnn = AveragedModel(dnn, avg_fn=avg_fn, mode='state_dict')
-        averaged_params = [torch.zeros_like(param) for param in dnn.state_dict().values()
+        averaged_dnn = AveragedModel(dnn, avg_fn=avg_fn, use_buffers=True)
```

Comment:
Do we have another test that covers the `use_buffers=False` case?
If not we should update this. You can use `@parametrize` decorator for the test to do that nicely.

Reply:
By default, `use_buffers` is `False`. This case is already covered by https://github.com/pytorch/pytorch/blob/0fdb90da5e1e7ec8f6becbdeecb84a3d855268ad/test/test_optim.py#L2450.

Reply:
Right, I guess we could deduplicate code. But this update sounds good.